receiver/sqlserverreceiver/README.md

```diff
@@ -14,29 +14,45 @@
 [sumo]: https://github.com/SumoLogic/sumologic-otel-collector
 <!-- end autogenerated section -->
 
-The `sqlserver` receiver grabs metrics about a Microsoft SQL Server instance using the Windows Performance Counters.
-Because of this, it is a Windows only receiver.
+The `sqlserver` receiver grabs metrics about a Microsoft SQL Server instance. The receiver works by either using the
```

Comment:
A lot of the complexity of this PR is enabling support on the non-Windows based OSs, while trying to keep Windows-specific functionality unchanged. I attempted to keep logic that was specific to windows in `_windows` files, logic specific to non-windows in `_others` files, and both in the generically named files. This leads to a bit of a challenge trying to follow some code, but I don't know of a better way given the nature of this receiver.

Happy to take suggestions on how to make this cleaner if anyone has any.